### PR TITLE
upgraded dokeysto_camltc and dokeysto_lz4 to 3.0.2

### PR DESCRIPTION
**Pull request by UnixJunkie**
dokeysto, dokeysto_lz4 and dokeysto_camltc are sister packages
(they all come from the same source tree); hence they should
be version synchronized. dokeysto is already at 3.0.2 in the opam-repository.

### CODE REVIEW

**Comment by UnixJunkie** [comment]
Some download failed in the Mac build:
https://travis-ci.org/ocaml/opam-repository/jobs/609028144?utm_medium=notification&utm_source=github_status
Maybe the CI continuous-integration/travis-ci/pr should be relaunched.

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+dev-repo: "git+https://github.com/UnixJunkie/dokeysto.git"
+license: "LGPL-2.1"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  [make "test_camltc"] {with-test}
+]
+depends: [
+  "ocaml"
+  "dune" {>= "1.0"}
+  "dokeysto"
```
Comment (kit-ty-kate) [inline packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]:
```suggestion
  "dokeysto" {= version}
```
Should they also match their current version then?

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+bug-reports: "https://github.com/UnixJunkie/dokeysto/issues"
+dev-repo: "git+https://github.com/UnixJunkie/dokeysto.git"
+license: "LGPL-2.1"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  [make "test_camltc"] {with-test}
+]
+depends: [
+  "ocaml"
+  "dune" {>= "1.0"}
```
Comment (kit-ty-kate) [inline packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]:
```suggestion
  "dune" {>= "1.11"}
```
According to your `dune-project` file, this package requires at least dune 1.11

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+bug-reports: "https://github.com/UnixJunkie/dokeysto/issues"
+dev-repo: "git+https://github.com/UnixJunkie/dokeysto.git"
+license: "LGPL-2.1"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  [make "test_lz4"] {with-test}
+]
+depends: [
+  "ocaml"
+  "dune" {>= "1.0" & < "2.0"}
```
Comment (kit-ty-kate) [inline packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]:
```suggestion
  "dune" {>= "1.11"}
```
According to your `dune-project` file, this package requires at least dune 1.11, also this package should now be compatible with dune 2.0

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "dokeysto"
+  "minicli" {>= "5.0.0"}
+  "lz4"
+  "dolog"
```
Comment (kit-ty-kate) [inline packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]:
```suggestion
```
I thought you removed this dependency

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+name: "dokeysto_lz4"
+authors: "Francois Berenger"
+maintainer: "unixjunkie@sdf.org"
+homepage: "https://github.com/UnixJunkie/dokeysto"
+bug-reports: "https://github.com/UnixJunkie/dokeysto/issues"
+dev-repo: "git+https://github.com/UnixJunkie/dokeysto.git"
+license: "LGPL-2.1"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  [make "test_lz4"] {with-test}
```
Comment (kit-ty-kate) [inline packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]:
```suggestion
  ["dune" "exec" "-p" name "-j" jobs "src/test_lz4.exe"] {with-test}
```
The make command hides a call to dune that doesn't have `-p` or `--profile`. This should not be used in releases

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam [hunk 6]
```diff
@@ -0,0 +1,23 @@
+opam-version: "2.0"
+authors: "Francois Berenger"
+maintainer: "unixjunkie@sdf.org"
+homepage: "https://github.com/UnixJunkie/dokeysto"
+bug-reports: "https://github.com/UnixJunkie/dokeysto/issues"
+dev-repo: "git+https://github.com/UnixJunkie/dokeysto.git"
+license: "LGPL-2.1"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  [make "test_camltc"] {with-test}
```
Comment (kit-ty-kate) [inline packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]:
```suggestion
  ["dune" "exec" "-p" name "-j" jobs "src/test_camltc.exe"] {with-test}
```
The make command hides a call to dune that doesn't have `-p` or `--profile`. This should not be used in releases

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+]
+depends: [
+  "ocaml"
+  "dune" {>= "1.0" & < "2.0"}
+  "dokeysto"
```
Comment (kit-ty-kate) [inline packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]:
```suggestion
  "dokeysto" {= version}
```
same question as above

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam (hunk 1 above)
Comment (UnixJunkie) [reply packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]: No, that's not necessary.

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam (hunk 5 above)
Comment (UnixJunkie) [reply packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]: The problem is that the Makefile's test target for that does more than just running the test exe.
```
test_lz4:
        \rm -f rwdb_lz4 rwdb_lz4.idx
        dune build _build/default/src/test_lz4.exe
        _build/default/src/test_lz4.exe
```
I'll comment that test line for the moment.

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam (hunk 6 above)
Comment (UnixJunkie) [reply packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]: Same here.

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam (hunk 7 above)
Comment (UnixJunkie) [reply packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]: This is not strictly necessary.

File: packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam (hunk 5 above)
Comment (kit-ty-kate) [reply packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]: I don't see what it does more than just removing file that are not present in the repository,  building and running the tests tbh. All of that are done by `dune exec`

Comment (UnixJunkie) [reply packages/dokeysto_lz4/dokeysto_lz4.3.0.2/opam]: ok, I'll put back your test line

File: packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam (hunk 6 above)
Comment (UnixJunkie) [reply packages/dokeysto_camltc/dokeysto_camltc.3.0.2/opam]: ok, I'll put back your test line

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**